broken-site/broken-site-impl/src/main/java/com/duckduckgo/brokensite/impl/RealBrokenSitePrompt.kt

```diff
@@ -43,16 +43,24 @@ class RealBrokenSitePrompt @Inject constructor(
 
     override suspend fun userDismissedPrompt() {
         if (!_featureEnabled) return
-        if (brokenSiteReportRepository.getDismissStreak() >= brokenSiteReportRepository.getMaxDismissStreak() - 1) {
-            brokenSiteReportRepository.resetDismissStreak()
-            val nextShownDate = brokenSiteReportRepository.getNextShownDate()
-            val newNextShownDate = currentTimeProvider.localDateTimeNow().plusDays(brokenSiteReportRepository.getDismissStreakResetDays().toLong())
 
+        val currentTimestamp = currentTimeProvider.localDateTimeNow()
+
+        brokenSiteReportRepository.addDismissal(currentTimestamp)
+
+        val dismissStreakCount = brokenSiteReportRepository.getDismissalCountBetween(
+            currentTimestamp.minusDays(brokenSiteReportRepository.getDismissStreakResetDays().toLong()),
+            currentTimestamp,
+        )
+
+        if (dismissStreakCount >= brokenSiteReportRepository.getMaxDismissStreak()) {
+            val newNextShownDate = currentTimestamp
```

Comment:
We don't need to update the next shown date with `getDismissStreakResetDays()` anymore, we just need to store the dismissal at this point, and then use `getDismissalCountBetween` when deciding whether we should show the prompt